app.js

```diff
@@ -48,32 +51,47 @@ const argv = yargs
 .env(true)
 .argv;
 
-if (argv.init || argv._.init || argv._.indexOf('init') > -1) {
-  init(argv);
-  log('Done!');
-  process.exit(0);
-}
-log(`Using local config directory: ${argv.config}`);
-const defaultConf = path.join(__dirname, 'conf');
+const main = () => {
+  if (argv.init || argv._.init || argv._.indexOf('init') > -1) {
+    init(argv);
+    log('Done!');
+    process.exit(0);
+  }
+  log(`Using local config directory: ${argv.config}`);
+  const defaultConf = path.join(__dirname, 'conf');
+  const conf = configHandler.loadConfig(defaultConf, argv, log);
+  if (!conf) {
+    process.exit(1);
+  }
+  // show css options:
+  if (argv.options || argv._.options || argv._.indexOf('options') > -1) {
+    reports.showOptions(conf);
+  // show css only:
+  } else if (argv.css) {
+    conf.cssExpression = argv.css;
+    reports.showCss(conf);
+  // dev mode will watch files and update when a change is made:
+  } else if (argv.mode === 'dev' || argv._.dev || argv._.indexOf('dev') > -1) {
+    dev.runDev(defaultConf, conf, argv, log);
+  // normal mode will run and output the new css/js dist directory:
+  } else {
+    if (argv.debug || argv._.indexOf('debug') > -1) {
+      log(JSON.stringify(conf, null, '  '));
+    }
+    run.runAll(conf, log);
+  }
+};
 
-const conf = configHandler.loadConfig(defaultConf, argv, log);
-if (!conf) {
-  process.exit(1);
-}
-// show css options:
-if (argv.options || argv._.options || argv._.indexOf('options') > -1) {
-  reports.showOptions(conf);
-// show css only:
-} else if (argv.css) {
-  conf.cssExpression = argv.css;
-  reports.showCss(conf);
-// dev mode will watch files and update when a change is made:
-} else if (argv.mode === 'dev' || argv._.dev || argv._.indexOf('dev') > -1) {
-  dev.runDev(defaultConf, conf, argv, log);
-// normal mode will run and output the new css/js dist directory:
-} else {
-  if (argv.debug || argv._.indexOf('debug') > -1) {
-    log(JSON.stringify(conf, null, '  '));
+const handleNotifier = (err, updateResults) => {
+  if (!err) {
+    if (updateResults.latest !== updateResults.current) {
+      log(`A new version of clientkit is available on npm! Current: ${updateResults.current}  Latest: ${updateResults.latest}`);
+    }
   }
-  run.runAll(conf, log);
-}
+  main();
+};
+
+updateNotifier({
+  callback: handleNotifier,
```

Comment:
For the most part, I think we want to check for an update directly and report right away and then move on to the main clientkit logic. The only reason to do it the other way is if we want to keep periodically checking when running clientkit in dev mode.  We can totally do it that way, but seemed like overkill to me.